docs/durable_execution/temporal.md

```diff
@@ -184,6 +184,81 @@ As the streaming model request activity, workflow, and workflow execution call a
 - To get data from the workflow call site or workflow to the event stream handler, you can use a [dependencies object](#agent-run-context-and-dependencies).
 - To get data from the event stream handler to the workflow, workflow call site, or a frontend, you need to use an external system that the event stream handler can write to and the event consumer can read from, like a message queue. You can use the dependency object to make sure the same connection string or other unique ID is available in all the places that need it.
 
+### Model Selection at Runtime
+
+[`Agent.run(model=...)`][pydantic_ai.agent.Agent.run] normally supports both model strings (like `'openai:gpt-5.2'`) and model instances. However, `TemporalAgent` does not support arbitrary model instances because they cannot be serialized for Temporal's replay mechanism.
+
+To use different models at runtime, you can:
+
+1. **Pre-register model instances**: Pass a dict of model instances to `TemporalAgent(models={...})`, then reference them by name or by passing the registered instance directly. If the wrapped agent doesn't have a model set, the first registered model will be used as the default.
+2. **Use model strings**: Model strings work as expected.
+3. **Use a provider factory**: For scenarios where you need to customize the provider (e.g., inject API keys from deps), pass a `provider_factory` to `TemporalAgent`.
```

Comment:
This doesn't explain that this is meant to be used alongside a model string, so maybe it works better as a sentence on item 2. And at that point, maybe we don't need a 2-item list at all anymore, and can make them paragraphs:

```
agent.run normally supports instances and model strings

with temporal, instances need to be preregistered

model strings just work, if you want to tweak the provider use the factory
```